app/code/Magento/InventoryIndexer/Plugin/InventoryApi/InvalidateAfterEnablingOrDisablingSourcePlugin.php

```diff
@@ -55,7 +55,7 @@ public function aroundSave(
         SourceInterface $source
     ) {
         $invalidationRequired = $this->isInvalidationRequiredForSource->execute(
-            $source->getSourceCode(),
+            (string)$source->getSourceCode(),
```

Comment:
why do you have null returned here?

Reply:
@maghamed, as we have validation test with "source code" = null and validation happens after this plugin, we get fatal error here.

Reply:
If `source_code` is empty we don't need to perform MySql query
Need to skip invalidation

Reply:
@naydav Done